R/arc-select.R

```diff
@@ -284,22 +274,42 @@ collect_layer <- function(x, n_max = Inf, token = Sys.getenv("ARCGIS_TOKEN"), ..
 
 # utility -----------------------------------------------------------------
 
-#' Check if x is a FeatureLayer or Table object
-#'
+#' Check if x is a FeatureLayer or Table class object
+#' @keywords internal
+obj_check_layer <- function(x,
+                            class = c("FeatureLayer", "Table"),
```

Comment:
I like the partitioning of `obj_check_layer` and `check_inherits_any()`. One issue for me, though, is that `obj_check_layer()` is not guaranteed to perform the check on FeatureLayer or Table because these can be easily overwritten in the `class` argument. Can we remove the class argument and hard code it into the call into check_inherits_any(). With the class argument it is a duplicate of check_inherits_any() but with a default class value.